compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -2220,6 +2220,12 @@ static bool simplifyISelectCompare(TR::Node *compare, TR::Simplifier *s)
       TR::NodeChecklist safetyVisited(s->comp());
       TR_ComparisonTypes compareType = TR::ILOpCode::getCompareType(compare->getOpCodeValue());
       bool isUnsignedCompare = TR::ILOpCode(compare->getOpCode()).isUnsignedCompare();
+
+      // do not try to simplify unsigned compare when it will yield a BadILOp with TR_cmpNE as its TR_ComparisonTypes
```

Comment:
As the call to `compareOpCode` no longer passes the `isUnsignedCompare` argument, this comment should probably not mentioned unsigned comparison any longer.  Maybe something like this:

```suggestion
      // do not try to simplify compare if it will yield a BadILOp with TR_cmpNE as its TR_ComparisonTypes
```

Reply:
Whoops, yes, I missed fixing up the language in the comment. Sorry about that - I've fixed it now.